File: apps/knowme-poc/packages/gen-ui-react/src/content-block.ts
// TJ-ARCH-MOB-001 compliant
// ContentBlock — the cross-platform UI contract, mirrored from gen_ui_types
// (Rust). serde emits { type: "text", text: "..." } etc (tag="type", camelCase).
// Keep in lockstep with crates/gen_ui_types/src/content_block.rs.

export type ContentBlock =
  | { type: "text"; text: string }
  | { type: "thinking"; text: string }
  | { type: "code"; language: string; code: string }
  | { type: "citation"; source: string; quote: string }
  | { type: "memory"; operation: string; key: string; value: string | null }
  | { type: "toolUse"; id: string; name: string; inputJson: string }
  | { type: "toolResult"; toolUseId: string; outputJson: string; isError: boolean }
  | { type: "skill"; name: string; status: string }
  | { type: "artifact"; id: string; kind: string; content: string }
  | { type: "image"; url: string | null; dataBase64: string | null; mime: string }
  | { type: "divider" };

File: apps/knowme-poc/rust/crates/gen_ui_types/src/content_block.rs
// TJ-ARCH-MOB-001 compliant
//! ContentBlock — the cross-platform UI contract. Every A2UI event maps to
//! exactly one variant. Dart/TS compilers enforce exhaustiveness at the match site.
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(tag = "type", rename_all = "camelCase")]
pub enum ContentBlock {
    Text { text: String },
    Thinking { text: String },
    Code { language: String, code: String },
    Citation { source: String, quote: String },
    Memory { operation: String, key: String, value: Option<String> },
    ToolUse { id: String, name: String, input_json: String },
    ToolResult { tool_use_id: String, output_json: String, is_error: bool },
    Skill { name: String, status: String },
    Artifact { id: String, kind: String, content: String },
    Image { url: Option<String>, data_base64: Option<String>, mime: String },
    Divider,
}
